tsl/test/sql/cagg_policy.sql

```diff
@@ -100,6 +149,106 @@ CREATE MATERIALIZED VIEW max_mat_view_date
+
+--Alter refresh_policy to make it open ended
+SELECT timescaledb_experimental.remove_policies('max_mat_view_date', false, 'policy_retention', 'policy_compression');
+SELECT timescaledb_experimental.alter_policies('max_mat_view_date', refresh_start_offset =>'-infinity');
+SELECT timescaledb_experimental.show_policies('max_mat_view_date');
+
+SELECT timescaledb_experimental.alter_policies('max_mat_view_date', refresh_end_offset =>'infinity', refresh_start_offset =>'5 days'::interval);
+SELECT timescaledb_experimental.show_policies('max_mat_view_date');
+
+--Cross policy checks
+-- Refresh and compression policies overlap
+SELECT timescaledb_experimental.add_policies('max_mat_view_date', compress_after => '20 days'::interval, drop_after => '25 days'::interval);
+SELECT timescaledb_experimental.alter_policies('max_mat_view_date', compress_after=> '4 days'::interval);
+SELECT timescaledb_experimental.show_policies('max_mat_view_date');
+
+-- Refresh and retention policies overlap
+SELECT timescaledb_experimental.alter_policies('max_mat_view_date', refresh_start_offset =>'5 days'::interval, drop_after=> '4 days'::interval);
+SELECT timescaledb_experimental.show_policies('max_mat_view_date');
+
+--Do not allow refreshed data to be deleted
+SELECT add_retention_policy('continuous_agg_max_mat_date', '25 days'::interval);
+SELECT timescaledb_experimental.alter_policies('max_mat_view_date', refresh_start_offset =>'25 days'::interval);
+SELECT remove_retention_policy('continuous_agg_max_mat_date');
+
+-- Remove one or more policy
+-- Code coverage: no policy names provided
+SELECT timescaledb_experimental.remove_policies('max_mat_view_date', false);
+
+-- Code coverage: incorrect name of policy
+SELECT timescaledb_experimental.remove_policies('max_mat_view_date', false, 'refresh_policy');
+
+SELECT timescaledb_experimental.remove_policies('max_mat_view_date', false, 'policy_refresh_continuous_aggregate', 'policy_compression');
+SELECT timescaledb_experimental.show_policies('max_mat_view_date');
+
+-- Add one policy
+SELECT timescaledb_experimental.add_policies('max_mat_view_date', refresh_start_offset => '15 day'::interval, refresh_end_offset => '1 day'::interval);
+SELECT timescaledb_experimental.show_policies('max_mat_view_date');
+
+-- Remove all policies
+SELECT * FROM timescaledb_experimental.policies;
+SELECT timescaledb_experimental.remove_all_policies(NULL); -- should fail
+SELECT timescaledb_experimental.remove_all_policies('continuous_agg_max_mat_date'); -- should fail
+SELECT timescaledb_experimental.remove_all_policies('max_mat_view_date', false);
+SELECT timescaledb_experimental.remove_all_policies('max_mat_view_date', false); -- should fail
+CREATE OR REPLACE FUNCTION custom_func(jobid int, args jsonb) RETURNS RECORD LANGUAGE SQL AS
+$$
+  VALUES($1, $2, 'custom_func');
+$$;
+SELECT add_job('custom_func','1h', config:='{"type":"function"}'::jsonb) AS job_id \gset -- inject custom job
```

Comment:
Sorry for being late to point this out, I think the comment should be on the line above as \gset takes this as a prefix

Reply:
Yeap. The added comment messed everything up...

Reply:
Fixed.